libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -2572,6 +2572,22 @@ bool AP_Mission::calc_rewind_pos(Mission_Command& rewind_cmd)
     return true;
 }
 
+//get next navigation command which also has valid location
+bool AP_Mission::get_next_nav_cmd_with_loc(Mission_Command& cmd) {
```

Comment:
Maybe?
```suggestion
bool AP_Mission::get_starting_nav_cmd_with_loc(Mission_Command& cmd) {
```
Open to there ideas, its not really the next command, or not always in anycase.

Reply:
It's not always the starting one, either: if `_restart` is true, it is the first command, while if `_restart` is not set, it is the next command. Maybe `get_nex_cmd_with_loc_after_restart`? That's long and still doesn't read clearly to me